### PR TITLE
Eliminate deprecated find_package(CUDA) from qmcpack

### LINKED ISSUE #3460: Transition CMake CUDA support to use FindCUDAToolkit
**Is your feature request related to a problem? Please describe.**

We currently use FindCUDA which is long deprecated. 

**Describe the solution you'd like**

Use modern FindCUDAToolkit instead.

**Describe alternatives you've considered**

None.

**Additional context**

Needed for modern compiler support - fixes are in FindCUDAToolkit.

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

This replaces find_package(CUDA) with first-class language support and find_package(CUDAToolkit).

On Ubuntu 20.04 and CUDA HPC SDK 21.7, I have to manually set the math libraries and headers. This appears to be fixed in CMake master https://gitlab.kitware.com/cmake/cmake/-/merge_requests/6405

On sulfur, I am encountering ABI issues between the old copy-on-write and small string optimized std::string. I assume it's a configuration issue on my part, but haven't tracked down where this is coming from and would like to run the tests here to confirm.

What documentation needs to be changed?

Fixes #3460

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Build related changes

### Does this introduce a breaking change?

- No

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes/No. Documentation has been added (if appropriate)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ######################################################################
 # CMake version and policies
 ######################################################################
-cmake_minimum_required(VERSION 3.15.0)
+cmake_minimum_required(VERSION 3.17.0)
```
Comment (ye-luo) [inline CMakeLists.txt]: Keep 3.15 and add a version guard before find_package(CUDAToolkit)

File: CMakeLists.txt [hunk 2]
```diff
@@ -260,6 +272,8 @@ endif()
 set(CMAKE_CXX_STANDARD ${QMC_CXX_STANDARD})
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CUDA_STANDARD ${QMC_CUDA_STANDARD})
```
Comment (ye-luo) [inline CMakeLists.txt:275]: Let us use the same QMC_CXX_STANDARD  for CUDA as well.

File: src/Platforms/CUDA/CMakeLists.txt [hunk 3]
```diff
@@ -13,9 +13,11 @@ set(CUDA_RT_SRCS CUDAfill.cpp CUDAallocator.cpp CUDAruntime.cpp)
...
 
 if(NOT QMC_CUDA2HIP)
-  cuda_add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
-  cuda_add_library(platform_cuda_LA ${CUDA_LA_SRCS})
-  cuda_add_cublas_to_target(platform_cuda_LA)
+  add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
+  add_library(platform_cuda_LA ${CUDA_LA_SRCS})
+  target_link_libraries(platform_cuda_LA PRIVATE CUDA::cublas)
+  target_include_directories(platform_cuda_runtime PUBLIC ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+  target_link_libraries(platform_cuda_runtime PRIVATE CUDA::cudart)
```
Comment (ye-luo) [inline src/Platforms/CUDA/CMakeLists.txt:20]: Use PUBLIC and remove the above line target_include_directories

File: src/Platforms/CUDA/CMakeLists.txt [hunk 4]
```diff
@@ -13,9 +13,11 @@ set(CUDA_RT_SRCS CUDAfill.cpp CUDAallocator.cpp CUDAruntime.cpp)
 set(CUDA_LA_SRCS cuBLAS_missing_functions.cu)
 
 if(NOT QMC_CUDA2HIP)
-  cuda_add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
-  cuda_add_library(platform_cuda_LA ${CUDA_LA_SRCS})
-  cuda_add_cublas_to_target(platform_cuda_LA)
+  add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
+  add_library(platform_cuda_LA ${CUDA_LA_SRCS})
+  target_link_libraries(platform_cuda_LA PRIVATE CUDA::cublas)
```
Comment (ye-luo) [inline src/Platforms/CUDA/CMakeLists.txt:18]: Add cusolver and remove CUDA_cusolver_LIBRARY in line 28.

File: src/Platforms/CUDA_legacy/CMakeLists.txt [hunk 5]
```diff
@@ -12,9 +12,9 @@
 set(CUDA_LEGACY_SRCS cuda_inverse.cu gpu_vector.cpp gpu_misc.cpp)
 
 if(NOT QMC_CUDA2HIP)
-  cuda_add_library(platform_cuda_legacy ${CUDA_LEGACY_SRCS})
-  cuda_add_cublas_to_target(platform_cuda_legacy)
-  target_link_libraries(platform_cuda_legacy PRIVATE ${CUDA_LIBRARIES})
+  add_library(platform_cuda_legacy ${CUDA_LEGACY_SRCS})
+  target_include_directories(platform_cuda_legacy PUBLIC ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
```
Comment (ye-luo) [inline src/Platforms/CUDA_legacy/CMakeLists.txt:16]: `target_link_libraries(platform_cuda_runtime PUBLIC CUDA::cudart) `

File: CMakeLists.txt [hunk 6]
```diff
@@ -14,6 +14,7 @@ cmake_policy(SET CMP0075 NEW)
 ######################################################################
 # QMCPACK project
 ######################################################################
+
```
Comment (ye-luo) [inline CMakeLists.txt:17]: Please remove the empty line

File: CMakeLists.txt [hunk 7]
```diff
@@ -662,38 +676,22 @@ if(QMC_CUDA OR ENABLE_CUDA)
...
-          CACHE STRING "CUDA architecture sm_XX")
-      set(CUDA_NVCC_FLAGS "-arch=${CUDA_ARCH};${CUDA_NVCC_FLAGS}")
-    endif(CUDA_NVCC_FLAGS MATCHES "arch")
-    include_directories(${CUDA_INCLUDE_DIRS})
+    if (CMAKE_VERSION VERSION_LESS 3.17.0)
+      message(FATAL_ERROR "QMC_CUDA or ENABLE_CUDA require CMake 3.17.0 or later")
+    endif()
+    enable_language(CUDA)
+    find_package(CUDAToolkit REQUIRED)
+    if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
```
Comment (ye-luo) [inline CMakeLists.txt:684]: Move this before line 682. Add a stopper for CUDA_ARCH and hints CMAKE_CUDA_ARCHITECTURES.

File: src/Platforms/CUDA_legacy/CMakeLists.txt (hunk 5 above)
Comment (quantumsteve) [reply src/Platforms/CUDA_legacy/CMakeLists.txt:16]: platform_cuda_runtime isn't part of this file. Did you mean ` target_link_libraries(platform_cuda_legacy PUBLIC CUDA::cublas)`?

Comment (ye-luo) [reply src/Platforms/CUDA_legacy/CMakeLists.txt:16]: Yes. Was a copy paste mistake.

**Comment by ye-luo** [comment]
Test this please

File: CMakeLists.txt (hunk 2 above)
Comment (quantumsteve) [reply CMakeLists.txt]: I removed CMAKE_CUDA_STANDARD and replaced it with QMC_CXX_STANDARD

File: src/Platforms/CUDA/CMakeLists.txt (hunk 3 above)
Comment (quantumsteve) [reply src/Platforms/CUDA/CMakeLists.txt]: changed

File: src/Platforms/CUDA_legacy/CMakeLists.txt (hunk 5 above)
Comment (quantumsteve) [reply src/Platforms/CUDA_legacy/CMakeLists.txt]: changed.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
In the case of NVHPC, I used CUDAToolkit_ROOT pointing to an standalone installation.

**Comment by prckent** [comment]
Please verify that this works with Summit's modules, software configuration etc.

**Comment by ye-luo** [comment]
@quantumsteve could you try again the current branch and paste what "errors about missing cublas headers." are. I'm afraid of missing/incorrect dependencies.

**Comment by ye-luo** [comment]
@prckent update to Summit build recipes has been included in the last commit.

**Comment by quantumsteve** [comment]
> @quantumsteve could you try again the current branch and paste what "errors about missing cublas headers." are. I'm afraid of missing/incorrect dependencies.

```
[125/663] Building CXX object src/QMCWaveFunctions/CMakeFiles/qmcwfs.dir/ElectronGas/ElectronGasOrbitalBuilder.cpp.o
FAILED: src/QMCWaveFunctions/CMakeFiles/qmcwfs.dir/ElectronGas/ElectronGasOrbitalBuilder.cpp.o 
/usr/bin/c++ -DADD_ -DH5_USE_16_API -DHAVE_CONFIG_H -DHAVE_EINSPLINE -DHAVE_LIBBOOST -DHAVE_LIBFFTW -DHAVE_LIBHDF5 -DOPENMP_NO_COMPLEX -DUSE_REAL_STRUCT_FACTOR -Drestrict=__restrict__ -I/home/svh/Documents/qmcpack/build/src -I/home/svh/Documents/qmcpack/src -I/home/svh/Documents/qmcpack/external_codes/boost_multi -I/home/svh/Documents/qmcpack/src/QMCWaveFunctions -I/home/svh/Documents/qmcpack/src/Particle -I/home/svh/Documents/qmcpack/src/Numerics -I/home/svh/Documents/qmcpack/src/io -I/home/svh/Documents/qmcpack/src/Containers -I/home/svh/Documents/qmcpack/src/Platforms -I/home/svh/Documents/qmcpack/src/Utilities -isystem /opt/nvidia/hpc_sdk/Linux_x86_64/21.7/cuda/11.4/include -isystem /usr/include/hdf5/serial -isystem /usr/include/libxml2 -fopenmp -finline-limit=1000 -fstrict-aliasing -funroll-all-loops -Wno-deprecated -Wvla -Wcomment -Wmisleading-indentation -Wmaybe-uninitialized -Wuninitialized -Wreorder -Wno-unknown-pragmas -Wno-sign-compare -Wsuggest-override -ffast-math -march=native -O3 -DNDEBUG -std=c++17 -MD -MT src/QMCWaveFunctions/CMakeFiles/qmcwfs.dir/ElectronGas/ElectronGasOrbitalBuilder.cpp.o -MF src/QMCWaveFunctions/CMakeFiles/qmcwfs.dir/ElectronGas/ElectronGasOrbitalBuilder.cpp.o.d -o src/QMCWaveFunctions/CMakeFiles/qmcwfs.dir/ElectronGas/ElectronGasOrbitalBuilder.cpp.o -c /home/svh/Documents/qmcpack/src/QMCWaveFunctions/ElectronGas/ElectronGasOrbitalBuilder.cpp
In file included from /home/svh/Documents/qmcpack/src/QMCWaveFunctions/Fermion/DelayedUpdateCUDA.h:19,
                 from /home/svh/Documents/qmcpack/src/QMCWaveFunctions/Fermion/DiracDeterminant.h:28,
                 from /home/svh/Documents/qmcpack/src/QMCWaveFunctions/Fermion/DiracDeterminantWithBackflow.h:24,
                 from /home/svh/Documents/qmcpack/src/QMCWaveFunctions/Fermion/SlaterDetWithBackflow.h:18,
                 from /home/svh/Documents/qmcpack/src/QMCWaveFunctions/ElectronGas/ElectronGasOrbitalBuilder.cpp:23:
/home/svh/Documents/qmcpack/src/Platforms/CUDA/cuBLAS.hpp:22:10: fatal error: cublas_v2.h: No such file or directory
   22 | #include <cublas_v2.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
```

**Comment by ye-luo** [comment]
```
diff --git a/src/Platforms/CUDA/CMakeLists.txt b/src/Platforms/CUDA/CMakeLists.txt
index f84c63ee1..ca1dd0092 100644
--- a/src/Platforms/CUDA/CMakeLists.txt
+++ b/src/Platforms/CUDA/CMakeLists.txt
@@ -16,7 +16,7 @@ if(NOT QMC_CUDA2HIP)
   add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
   target_link_libraries(platform_cuda_runtime PUBLIC CUDA::cudart)
   add_library(platform_cuda_LA ${CUDA_LA_SRCS})
-  target_link_libraries(platform_cuda_LA PRIVATE CUDA::cublas CUDA::cusolver)
+  target_link_libraries(platform_cuda_LA PUBLIC CUDA::cublas CUDA::cusolver)
 else()
   hip_add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
   target_link_libraries(platform_cuda_LA PUBLIC platform_rocm_LA)
@@ -24,4 +24,4 @@ else()
   target_link_libraries(platform_cuda_runtime PUBLIC platform_rocm_runtime)
 endif()
 
-target_link_libraries(platform_cuda_LA PRIVATE platform_cuda_runtime)
+target_link_libraries(platform_cuda_LA PUBLIC platform_cuda_runtime)
yeluo@epyc-server:~/opt/cleanup/qmcpack$ 
```
Try this.

**Comment by ye-luo** [comment]
Regarding docs. CUDA_ARCH to CMAKE_CUDA_ARCHITECTURES is the most notable change.

**Comment by prckent** [comment]
Thanks @ye-luo  Re: Summit. I was expecting more weirdness.

The manual might need updating.

The scripts for the nightlies presumably will as well, but we (I) can fix those after this is in.

**Comment by quantumsteve** [comment]
> ```
> diff --git a/src/Platforms/CUDA/CMakeLists.txt b/src/Platforms/CUDA/CMakeLists.txt
> index f84c63ee1..ca1dd0092 100644
> --- a/src/Platforms/CUDA/CMakeLists.txt
> +++ b/src/Platforms/CUDA/CMakeLists.txt
> @@ -16,7 +16,7 @@ if(NOT QMC_CUDA2HIP)
>    add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
>    target_link_libraries(platform_cuda_runtime PUBLIC CUDA::cudart)
>    add_library(platform_cuda_LA ${CUDA_LA_SRCS})
> -  target_link_libraries(platform_cuda_LA PRIVATE CUDA::cublas CUDA::cusolver)
> +  target_link_libraries(platform_cuda_LA PUBLIC CUDA::cublas CUDA::cusolver)
>  else()
>    hip_add_library(platform_cuda_runtime ${CUDA_RT_SRCS})
>    target_link_libraries(platform_cuda_LA PUBLIC platform_rocm_LA)
> @@ -24,4 +24,4 @@ else()
>    target_link_libraries(platform_cuda_runtime PUBLIC platform_rocm_runtime)
>  endif()
>  
> -target_link_libraries(platform_cuda_LA PRIVATE platform_cuda_runtime)
> +target_link_libraries(platform_cuda_LA PUBLIC platform_cuda_runtime)
> yeluo@epyc-server:~/opt/cleanup/qmcpack$ 
> ```
> 
> Try this.

No change. I was able to build with the following change to FindCUDAToolkit.cmake

```
diff --git a/Modules/FindCUDAToolkit.cmake b/Modules/FindCUDAToolkit.cmake
index 92042d23c4..6f0fa47666 100644
--- a/Modules/FindCUDAToolkit.cmake
+++ b/Modules/FindCUDAToolkit.cmake
@@ -853,7 +853,7 @@ if(CUDAToolkit_FOUND)
 
     if (NOT TARGET CUDA::${lib_name} AND CUDA_${lib_name}_LIBRARY)
       add_library(CUDA::${lib_name} UNKNOWN IMPORTED)
-      target_include_directories(CUDA::${lib_name} SYSTEM INTERFACE "${CUDAToolkit_INCLUDE_DIRS}")
+      target_include_directories(CUDA::${lib_name} SYSTEM INTERFACE "${CUDAToolkit_INCLUDE_DIRS}" "${CUDAToolkit_INCLUDE_DIRS}/../../../../../math_libs/include")
       set_property(TARGET CUDA::${lib_name} PROPERTY IMPORTED_LOCATION "${CUDA_${lib_name}_LIBRARY}")
       foreach(dep ${arg_DEPS})
         if(TARGET CUDA::${dep})
```

**Comment by ye-luo** [comment]
I was confused by your earlier message. Here is my understanding.

With all the released CMake, CUDA accelerated math libraries shipped inside NVHPC are not well handled by FindCUDAToolkit. In this case, I recommend using CUDAToolkit_ROOT pointing to a standalone CUDA toolkit installation. It is not worth teaching FindCUDAToolkit how to use NVHPC internally shipped CUDA toolkit.
`${CUDAToolkit_INCLUDE_DIRS}/../../../../../math_libs/include` is only valid for NVHPC internally shipped CUDA toolkit. It should not be needed when CUDAToolkit_ROOT pointing to a standalone CUDA toolkit installation.

Once 3.22 is release, NVHPC shipped CUDA toolkit will be recognized by FindCUDAToolkit.

**Comment by quantumsteve** [comment]
> Once 3.22 is release, NVHPC shipped CUDA toolkit will be recognized by FindCUDAToolkit.

CMake master branch is still broken. I [created an issue on CMake gitlab](https://gitlab.kitware.com/cmake/cmake/-/issues/22706).

I'll reinstall the standalone toolkit locally so these NVHPC issues don't affect me.

File: CMake/ClangCompilers.cmake [hunk 8]
```diff
@@ -19,6 +19,10 @@ if(QMC_OMP)
         CACHE STRING "Offload target architecture")
     set(OPENMP_OFFLOAD_COMPILE_OPTIONS "-fopenmp-targets=${OFFLOAD_TARGET}")
 
+    if(NOT DEFINED OFFLOAD_ARCH AND OFFLOAD_TARGET MATCHES "nvptx64" AND DEFINED CMAKE_CUDA_ARCHITECTURES)
+      set(OFFLOAD_ARCH sm_${CMAKE_CUDA_ARCHITECTURES})
```
Comment (quantumsteve) [inline CMake/ClangCompilers.cmake:23]: @ye-luo Do we allow `CMAKE_CUDA_ARCHITECTURES` be a list of integers?

Comment (ye-luo) [reply CMake/ClangCompilers.cmake:23]: CMAKE_CUDA_ARCHITECTURES should be allowed as the name indicates.
LLVM doesn't support offload to multiple architectures but one.
We might need to skip this fallback if CMAKE_CUDA_ARCHITECTURES is a list of integers. Is this the check you would like to add?

**Comment by markdewing** [comment]
This change appears to require more effort to get the right CUDA binaries and include files.
My laptop is Ubuntu 18.04, and the default cuda install is 9.1.   I have newer CUDA versions installed in `/usr/local/cuda-*.*`
I'm using a recent mainline build of llvm for the offload compiler.

Prior to this change, I used the configuration (just the CUDA relevant ones are shown):
```
-DCUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-11.3 \
-DCUDA_HOST_COMPILER=`which gcc` \
-DCMAKE_CXX_FLAGS="-I/usr/local/cuda-11.3/include" \
-DENABLE_OFFLOAD=1 \
-DENABLE_CUDA=1 \
```

To use this PR, the configuration options are
```
-DCMAKE_CUDA_ARCHITECTURES=61 \
-DCUDAToolkit_ROOT=/usr/local/cuda-11.3 \
-DCMAKE_CXX_FLAGS="-I/usr/local/cuda-11.3/include" \
-DENABLE_OFFLOAD=1 \
-DENABLE_CUDA=1 \
```

The first problem was it picked up the system `nvcc`.  I changed the path to include `/usr/local/cuda-11.3/bin`, and it picked up the right `nvcc`.
But now it appears to be getting the wrong CUDA include file, as in the error below.   The right cuda include directory is already given using `CMAKE_CXX_FLAGS` - I'm not sure how to get it to use the right one.

```
In file included from /home/mdewing/physics/codes/qmcpack/github/qmcpack/src/QMCWaveFunctions/tests/test_cuBLAS_LU.cpp:18:
/home/mdewing/physics/codes/qmcpack/github/qmcpack/src/Platforms/CUDA/cuBLAS.hpp:253:10: error: no matching function for call to 'cublasSgemmBatched'
  return cublasSgemmBatched(handle, transa, transb, m, n, k, alpha, A, lda, B, ldb, beta, C, ldc, batchCount);
         ^~~~~~~~~~~~~~~~~~
/usr/include/cublas_api.h:2308:39: note: candidate function not viable: 8th argument ('const float *const *') would lose const qualifier
CUBLASAPI cublasStatus_t CUBLASWINAPI cublasSgemmBatched (cublasHandle_t handle,
                                      ^
```


I'm not certain the previous configuration (`develop`) works because with the llvm build I have, it crashes in the compiler during compilation.

File: CMake/ClangCompilers.cmake (hunk 8 above)
Comment (quantumsteve) [reply CMake/ClangCompilers.cmake:27]: I added a check that the list length is 1

**Comment by PDoakORNL** [comment]
I don't think we should have to set the cuda include path by hand, infact that breaks the way the CUDA::xxx targets should work.  If a targets dependency on the CUDA targets is expressed properly through its target_link_libraries and possibly its link libaries target_link_libraries the include directory should match the version linked without fail.

**Comment by ye-luo** [comment]
@markdewing does patch https://github.com/QMCPACK/qmcpack/pull/3492#issuecomment-931406112 work?

**Comment by quantumsteve** [comment]
@markdewing 
What is the value of `CUDAToolkit_INCLUDE_DIRS`?

Does changing `find_package(CUDAToolkit REQUIRED)` to `find_package(CUDAToolkit 11.3 REQUIRED)` make any difference?

File: CMake/ClangCompilers.cmake (hunk 8 above)
Comment (ye-luo) [reply CMake/ClangCompilers.cmake]: I also made NVHPC working with CMAKE_CUDA_ARCHITECTURES  as a list of integers.

**Comment by markdewing** [comment]
Turns out it was the added include directory that was causing the problem. (`-DCMAKE_CXX_FLAGS="-I/usr/local/cuda-11.3/include"`)  It was something I added sometime to fix some problem (NVTX api maybe?), and just left it there.   It still seems odd that it's presence would have such a large effect.

**Comment by ye-luo** [comment]
Test this please

File: config/build_olcf_summit_Clang.sh [hunk 9]
```diff
@@ -32,28 +32,29 @@ module load llvm/main-20210811-cuda10.1
 TYPE=Release
 Compiler=Clang
 
-source_folder=..
+source_folder=~/opt/qmcpack
```
Comment (ye-luo) [inline config/build_olcf_summit_Clang.sh:35]: I need to change this before merging.

**Comment by ye-luo** [comment]
This PR is ready to be merged except I need to fix a minor scripting issue. If there is no other concerns, I'd like to merge it this afternoon.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Comment by prckent** [comment]
Is it expected that any changes will be needed to the nightly scripts?

**Comment by ye-luo** [comment]
> Is it expected that any changes will be needed to the nightly scripts?
```
yeluo@ryzen-box:~/opt/qmcpack/tests/test_automation/nightly_test_scripts$ grep CUDA_ *
nightly_anl_bora.sh:    CTEST_FLAGS="$CTEST_FLAGS -DQMC_OPTIONS='-DENABLE_OFFLOAD=ON;-DUSE_OBJECT_TARGET=ON;-DCUDA_HOST_COMPILER=`which gcc`;-DCUDA_PROPAGATE_HOST_FLAGS=OFF;-DQMC_NIO_MAX_SIZE=16'"
nightly_anl_bora.sh:    CTEST_FLAGS="$CTEST_FLAGS -DENABLE_CUDA=1 -DCUDA_ARCH=sm_61"
nightly_anl_bora.sh:    CTEST_FLAGS="$CTEST_FLAGS -DENABLE_CUDA=1 -DCUDA_ARCH=sm_61 -DQMC_OPTIONS='-DQMC_NIO_MAX_SIZE=16'"
nightly_anl_bora.sh:    CTEST_FLAGS="$CTEST_FLAGS -DQMC_CUDA=1 -DCUDA_ARCH=sm_61 -DQMC_OPTIONS='-DQMC_NIO_MAX_SIZE=16'"
nightly_ornl.sh:        CTCFG="$CTCFG -DCMAKE_CXX_FLAGS=-Wno-unknown-cuda-version -DQMC_OPTIONS='-DENABLE_OFFLOAD=ON;-DUSE_OBJECT_TARGET=ON;-DENABLE_CUDA=ON;-DCUDA_ARCH=sm_70;-DCUDA_HOST_COMPILER=`which gcc`'"
nightly_ornl.sh:#export CUDA_VISIBLE_DEVICES=
weekly_ornl.sh:        CTCFG="$CTCFG -DCMAKE_CXX_FLAGS=-Wno-unknown-cuda-version -DQMC_OPTIONS='-DENABLE_OFFLOAD=ON;-DUSE_OBJECT_TARGET=ON;-DENABLE_CUDA=ON;-DCUDA_ARCH=sm_70;-DCUDA_HOST_COMPILER=`which gcc`'"
weekly_ornl.sh:#export CUDA_VISIBLE_DEVICES=
```
Needs update as
CUDA_ARCH -> CMAKE_CUDA_ARCHITECTURES
CUDA_HOST_COMPILER -> CMAKE_CUDA_HOST_COMPILER

**Comment by prckent** [comment]
Scripts are now updated. Tests that actually run will be limited due to the recent+unsolved by me cmake/ctest/scripting breakage.

**Pull request merged**